### PR TITLE
:bug: simple / isInitialized / default defaultData 

### LINKED ISSUE #159: factory / simple / selector isInitialized don't work without defaultData
https://github.com/alakarteio/k-redux-factory/blob/master/src/types/simple/simple.selectors.js#L9

### PR DESCRIPTION
**Pull request by guillaumecrespel**
fix #159

### CODE REVIEW

**Comment by fabienjuif** [comment]
Can you add a test (that you trigger before your fix to make sur this is KO) please?

File: src/factories.js [hunk 1]
```diff
@@ -34,16 +34,15 @@ const getWrappedStore = (middlewares = {}) => (options = {}) => {
...
   }
 
   const result = Object.assign(
-    reducer({ ...middlewares, engine: typeConfig.middlewares })(key)(prefix)(name)(innerDefaultData),
+    reducer({ ...middlewares, engine: typeConfig.middlewares })(key)(prefix)(name)(innerOptions.defaultData),
 
-    // type (debug purpose)
-    { krfType: type },
+    // options (debug purpose)
+    { krfType: type, options: innerOptions },
```
Comment (fabienjuif) [inline src/factories.js]: I prefer not to set to much things on our reducer (that are functions)

File: src/factories.spec.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { simple, keyValue } from './factories'
+
+describe('factories', () => {
+  it('should use options of object type', () => expect(simple.object('object').options).toMatchSnapshot())
+  it('should use options of specific string type', () => expect(simple.string('string').options).toMatchSnapshot())
+  it('should use options of specific array type', () => expect(simple.array('array').options).toMatchSnapshot())
+  it('should use options of specific boolean type', () => expect(simple.bool('boolean').options).toMatchSnapshot())
+  it('should use options of specific keyValue type', () => expect(keyValue('keyValue').options).toMatchSnapshot())
+
+  it('should use overwrited options of object type', () => expect(simple.object({ name: 'object', defaultData: { o: 'test' } }).options).toMatchSnapshot())
```
Comment (fabienjuif) [inline src/factories.spec.js]: Your PR  is not about options but about defaultData (and in a higher level, about isInitialized)
1. Make isInitialized bugguy again
2. Find a test that show it bugs
3. Fix this test

Only that

File: src/factories.js [hunk 3]
```diff
@@ -34,15 +34,14 @@ const getWrappedStore = (middlewares = {}) => (options = {}) => {
...
+  if (defaultData === undefined && subType) {
+    innerOptions.defaultData = simpleDefaultData[subType]
   }
 
   const result = Object.assign(
-    reducer({ ...middlewares, engine: typeConfig.middlewares })(key)(prefix)(name)(innerDefaultData),
+    reducer({ ...middlewares, engine: typeConfig.middlewares })(key)(prefix)(name)(innerOptions.defaultData),
 
-    // type (debug purpose)
+    // options (debug purpose)
```
Comment (fabienjuif) [inline src/factories.js]: you can rollback this line

Comment (guillaumecrespel) [reply src/factories.js]:
```suggestion
    // type (debug purpose)
```

**Pull request merged**